partner_event/models/event_registration.py

```diff
@@ -25,29 +26,43 @@ def _prepare_partner(self, vals):
             "phone": vals.get("phone", False),
         }
 
-    @api.model
-    def create(self, vals):
-        if not vals.get("attendee_partner_id") and vals.get("email"):
-            Partner = self.env["res.partner"]
-            Event = self.env["event.event"]
-            # Look for a partner with that email
-            email = vals.get("email").replace("%", "").replace("_", "\\_")
-            attendee_partner = Partner.search([("email", "=ilike", email)], limit=1)
-            event = Event.browse(vals["event_id"])
-            if attendee_partner:
-                vals["name"] = vals.setdefault("name", attendee_partner.name)
-                vals["phone"] = vals.setdefault("phone", attendee_partner.phone)
-            elif event.create_partner:
-                # Create partner
-                attendee_partner = Partner.sudo().create(self._prepare_partner(vals))
-            vals["attendee_partner_id"] = attendee_partner.id
-        return super(EventRegistration, self).create(vals)
+    @api.model_create_multi
+    def create(self, vals_list):
+        for values in vals_list:
+            if not values.get("attendee_partner_id") and values.get("email"):
+                Partner = self.env["res.partner"]
+                Event = self.env["event.event"]
+                # Look for a partner with that email
+                email = values.get("email").replace("%", "").replace("_", "\\_")
+                attendee_partner = Partner.search([("email", "=ilike", email)], limit=1)
+                event = Event.browse(values["event_id"])
+                if attendee_partner:
+                    values["name"] = (
+                        not values.get("name")
+                        and attendee_partner.name
+                        or values.get("name")
+                    )
+                    values["phone"] = (
+                        not values.get("phone")
+                        and attendee_partner.phone
+                        or values.get("phone")
+                    )
+                    values["mobile"] = (
+                        not values.get("mobile")
+                        and attendee_partner.mobile
+                        or values.get("mobile")
+                    )
```

Comment:
Simplify this expression:
```suggestion
                    for field in {"name", "phone", "mobile"}:
                        vals[field] = vals.get(field) or attendee[field]
```

Reply:
Ok it's already simplified